### DISCUSSION TITLE
Download data from OECD

### DISCUSSION THREAD

**Discussion by lz1nwm** [amattioc/SDMX#269] [Q&A] [answered]
Hi, I want to make request like this in R:
[sdmx.oecd.org/public/rest/v2/data/dataflow/OECD.ECO.MAD/DSD_EO@DF_EO/1.1/*.CB.A](https://sdmx.oecd.org/public/rest/v2/data/dataflow/OECD.ECO.MAD/DSD_EO@DF_EO/1.1/*.CB.A)

I tried this code but it doesn't work:
`getTimeSeriesTable2(provider='OECD_SDMXV3', dataflow = 'OECD.ECO.MAD', key = 'DSD_EO@DF_EO/1.1/*.CB.A')`

I also tried with `OECD_NEW`, but then I get `Connection error. Code:400 Message:Invalid structure: data`

**Comment by amattioc** [comment, marked as answer]
Hi @lz1nwm , this is the correct query:

`getTimeSeriesTable2(provider='OECD_SDMXV3', dataflow = 'DSD_EO@DF_EO', key = '*.CB.A')`

the dataflow id is **DSD_EO@DF_EO** , the connectors manage the agency (OECD.ECO.MAD) and version automatically, so they do not have to be provided.

You may also want to try the new filter based query style:

`getTimeSeries2(provider='OECD_SDMXV3', dataflow='DSD_EO@DF_EO', filter='c[MEASURE]=CB&c[FREQ]=A');`

Both query types  take a while to complete (13k time series)

Best, Attilio

**Reply by lz1nwm** [reply]
Thank you @amattioc ! 

I have one more question. The following code should return four series - two indicators for two countries, but instead it returns 104,067 rows (...the whole dataset?). Why is that?

```{r}
geo <- c('AUS,AUT')
measuere <- c('GDPV_ANNPCT,CPI_YTYPCT')
dt.oecd <- getTimeSeriesTable2(provider='OECD_SDMXV3', 
                               dataflow = 'DSD_EO@DF_EO', 
                               filter = paste0('c[REF_AREA]=',geo,'&c[MEASURE]=',measuere,'&c[FREQ]=A'), 
                               start = "2022")
```

**Comment by amattioc** [comment]
Hi, you are right. OECD v2 does not support filtering on component value yet, I forgot it. They will add it probably at a later time. Anyway the query contruct is correct, you can see it in a similar query on a test provider:

`result <- getTimeSeries2(provider='DEMO_SDMXV3', dataflow='EXR', filter='c[FREQ]=A&c[CURRENCY]=ARS,AUD&c[EXR_SUFFIX]=A,E');`
